test/jdk/com/sun/jdi/lib/jdb/Debuggee.java

```diff
@@ -127,33 +125,25 @@ public ProcessBuilder prepare() {
         }
 
         public Debuggee launch(String name) {
-            return new Debuggee(prepare(), name, waitForPortPrint, expectedOutputBeforeThrow);
+            return new Debuggee(prepare(), name, !onthrow.isEmpty(), expectedOutputBeforeThrow);
         }
         public Debuggee launch() {
             return launch("debuggee");
         }
     }
 
     // starts the process, waits for "Listening for transport" output and detects transport/address
-    private Debuggee(ProcessBuilder pb, String name, boolean waitForPortPrint, String expectedOutputBeforeThrow) {
+    private Debuggee(ProcessBuilder pb, String name, boolean hasOnThrow, String expectedOutputBeforeThrow) {
```

Comment:
Launcher prepares command line and address detection logic depends on on command line, so I think it would be better to make
```suggestion
    private Debuggee(ProcessBuilder pb, String name, Function<String, JDWP.ListenAddress> addressDetector) {
```
then warm-up predicate in the ctor uses `listenAddress[0] = addressDetector.apply(s)`

Launcher.launch() can pass JDWP::parseListenAddress for usual cases and implement the detector for onthrow case